traits/ctraits.c

```diff
@@ -4106,6 +4106,10 @@ post_setattr_trait_python(
     trait_object *trait, has_traits_object *obj, PyObject *name,
     PyObject *value)
 {
+    if (trait->py_post_setattr == Py_None) {
```

Comment:
Ah, good catch. But I think this isn't quite the right solution. I think we want a simple two-way invariant on the `py_post_setattr` field: either it's `NULL`, or it contains a valid callable. Expanding to three possible states (`NULL`, `Py_None` or callable) makes the code more complicated.

So instead of this, I think what we should be doing is turning a `None` input in the setter into a `NULL` in the field.